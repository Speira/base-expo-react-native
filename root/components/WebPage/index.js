import React from 'react'
import PropTypes from 'prop-types'
import BaseWebPage from './style'

/**
 * WebPage
 * @component
 * @desc ::: View for Web
 *
 */
function WebPage(props) {
  const { hide, uri } = props
  return hide ? null : <BaseWebPage source={{ uri }} />
}
WebPage.defaultProps = {
  hide: false,
}
WebPage.propTypes = {
  hide: PropTypes.bool,
  uri: PropTypes.string.isRequired,
}

export default WebPage
