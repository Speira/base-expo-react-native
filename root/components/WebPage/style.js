import styled from 'styled-components/native'
import { WebView } from 'react-native-webview'

/**
 * BaseWebPage
 * @component
 *
 */
const BaseWebPage = styled(WebView)``

export default BaseWebPage
