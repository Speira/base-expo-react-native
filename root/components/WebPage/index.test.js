import React from 'react'
import renderer from 'react-test-renderer'
import WebPage from './index'

test('render WebPage', () => {
  const tree = renderer.create(<WebPage uri="http://test.com" />).toJSON()
  expect(tree).toMatchSnapshot()
})
