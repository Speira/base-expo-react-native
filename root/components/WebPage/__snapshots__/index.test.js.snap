// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render WebPage 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <RNCWebView
    cacheEnabled={true}
    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
    injectedJavaScriptForMainFrameOnly={true}
    javaScriptEnabled={true}
    messagingEnabled={false}
    onContentProcessDidTerminate={[Function]}
    onHttpError={[Function]}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={
      Object {
        "uri": "http://test.com",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Array [
          Object {},
        ],
      ]
    }
    textInteractionEnabled={true}
    useSharedProcessPool={true}
  />
</View>
`;
